packages/big-design/src/components/Table/Row/Row.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { memo, useContext, useState } from 'react';
+
+import { Checkbox } from '../../Checkbox';
+import { TableContext, TableSectionContext } from '../context';
+import { TableCell } from '../Cell/Cell';
+
+import { StyledTableRow } from './styled';
+
+export interface TableRowProps extends React.TableHTMLAttributes<HTMLTableRowElement> {
+  selected?: boolean;
+  verticalAlign?: 'top' | 'bottom' | 'middle';
+  onRowSelect?(selected: boolean): void;
+}
+
+export const TableRow: React.FC<TableRowProps> = memo(({ children, selected, onRowSelect, ...props }) => {
+  const tableContext = useContext(TableContext);
+  const tableSectionContext = useContext(TableSectionContext);
+  const [isSelected, setIsSelected] = useState(selected);
```

Comment:
I was under the impression `state` shouldn't be set from a `prop` received. Is this a pattern that has changed with hooks?